utils/keepassxc-kdewallet

```diff
@@ -12,16 +14,14 @@ function daemon_main {
     sleep 1
   done
 
-  # fetch KeePass database passwords from kdewallet
   declare -A DBs
-  for DBPATH in $KDBX_SEARCH; do
-    [[ -L "$DBPATH" ]] && DBPATH=$(readlink --canonicalize "$DBPATH")
-    DBs[$DBPATH]=$(qdbus org.kde.kwalletd5 /modules/kwalletd5 org.kde.KWallet.readPassword "$handle" "Passwords" "$DBPATH" "$PROG")
+  for DBPATH in $(ls -r $KDBX_SEARCH); do
+    DBs[$(realpath $DBPATH)]=$(qdbus org.kde.kwalletd5 /modules/kwalletd5 org.kde.KWallet.readPassword "$handle" "Passwords" "${DBPATH##*/}" "$PROG")
   done
 
-  # launch keepassx
+  # launch real keepassxc
   IFS=$'\n\n\n'
-  keepassx --pw-stdin "${!DBs[@]}" <<<"${DBs[*]}" &
+  "$KEEPASSXC" --pw-stdin "${!DBs[@]}" <<<"${DBs[*]}" &
```

Comment:
Why don't we just use dbus here?
1. Start KeePassXC
2. Wait for settle
3. Send dbus commands (https://github.com/keepassxreboot/keepassxc/blob/develop/src/gui/org.keepassxc.KeePassXC.MainWindow.xml)

Reply:
when i wrote this dbus commands were a feature request, not an implementation.  it probably makes sense to redo this with dbus but i've used this script for years with multiple databases before 2.4.0 came along.  it was also easy to replicate the functionality for MacOS in https://github.com/jkt628/keepassxc/blob/pw-stdin/utils/keepassxc-keychain and maintain alignment.

Reply:
Perhaps we can provide both methods because dbus would allow you to use multiple key files.